wordpress-mobile/WordPress-iOS
# Don't save synced blogs after signed out

Fixes #20928.

This crash (or the failed assertion `NSParameterAssert(account.managedObjectContext == context)`) happens because syncing blogs operation (which is kicked off from the site picker screen) continues to run even though the app has signed out. When the syncing blogs API request(s) come back, the account is deleted from the database and the assertion fails because `account` is `nil`.

## Test Instructions

See https://github.com/wordpress-mobile/WordPress-iOS/issues/20928#issuecomment-1608570438 for steps to reproduce this bug. Verify the crash is no longer reproducible. And you should see a "Can't find the account. User may have signed out." log on Xcode console output.

You can also reproduce the issue by commenting out the `BlogService` changes and running the new unit test added in this PR.

## Regression Notes
1. Potential unintended areas of impact
None.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
What's described in the Test Instructions.

3. What automated tests I added (or what prevented me from doing so)
One unit test that can be used to reproduce the issue.

PR submission checklist:


UI Changes testing checklist: N/A
